worlds/sc2/MissionOrders.py

```diff
@@ -0,0 +1,389 @@
+            FillMission(MissionPools.HARD, [MissionConnection(2, SC2Campaign.NCO)], "_3", completion_critical=True),
+            FillMission(MissionPools.FINAL, [MissionConnection(3, SC2Campaign.NCO)], "_3", completion_critical=True),
+        ]
+    }
+
+
+smooth_difficulty = [MissionPools.EASY,
+                     MissionPools.MEDIUM, MissionPools.MEDIUM,
+                     MissionPools.HARD, MissionPools.HARD,
+                     MissionPools.VERY_HARD]
+max_difficulty = len(smooth_difficulty) - 1
+
+
+def make_golden_path(world: World, num_missions: int) -> List[FillMission]:
+    chain_name_options = ['Mar Sara', 'Char', 'Kaldir', 'Zerus', 'Skygeirr Station',
+                   'Dominion Space', 'Korhal', 'Aiur', 'Shakuras', 'Ulnar']
+    world.random.shuffle(chain_name_options)
+
+    first_chain = chain_name_options.pop()
+    first_mission = FillMission(MissionPools.STARTER, [MissionConnection(-1, SC2Campaign.GLOBAL)], first_chain,
+                                     completion_critical=True)
+
+    class Campaign:
+        def __init__(self, first_mission: FillMission, missions_remaining: int):
+            self.mission_order = [first_mission]
+            self.counter = 0
+            self.last_mission_in_chain = [0]
+            self.chain_names = [first_mission.category]
+            self.missions_remaining = missions_remaining
+            self.padding = 0
+
+        def add_mission(self, chain: int, difficulty: int, required_missions: int = 0):
+            if self.missions_remaining == 0 and difficulty is not MissionPools.FINAL:
+                return
+            self.counter += 1
+            if self.mission_order[self.last_mission_in_chain[chain]].number == required_missions or required_missions <= 1:
+                required_missions = 0
+            mission_connections = [MissionConnection(self.last_mission_in_chain[chain], SC2Campaign.GLOBAL)]
+            padding = 0
+            if self.last_mission_in_chain[chain] == self.last_mission_in_chain[0]:
+                # Adding padding to the start of new chains
+                if chain != 0:
+                    padding = self.padding
+            else:
+                # Requiring main chain progress for optional chains
+                mission_connections.append(MissionConnection(self.last_mission_in_chain[0], SC2Campaign.GLOBAL))
+            self.mission_order.append(FillMission(
+                difficulty,
+                mission_connections,
+                self.chain_names[chain],
+                number=required_missions,
+                completion_critical=chain == 0,
+                ui_vertical_padding=padding
+            ))
+            self.last_mission_in_chain[chain] = self.counter
+            if chain == 0:
+                self.padding += 1
+            self.missions_remaining -= 1
+
+    campaign = Campaign(first_mission, num_missions - 2)
+    current_required_missions = 0
+    main_chain = 0
```

Comment:
Recommend renaming to `main_chain_length` or similar.

Reply:
Refactored.